### ISSUE TITLE
Disabling rules on a per-line basis does not work inside array constructs

### ISSUE THREAD

**Issue by zachriggle** [koalaman/shellcheck#2283]
#### For bugs
- Rule Id SC2206
- My shellcheck version "online"
- [x] The rule's wiki page does not already cover this (e.g. https://shellcheck.net/wiki/SC2086)
- [x] I tried on https://www.shellcheck.net/ and verified that this is still a problem on the latest commit

#### Here's a snippet or screenshot that shows the problem:

There are no errors fitted if the disable line is moved to just after the shebang, or to before the declaration of the array.

*Broken*
```sh
#!/usr/bin/env bash
true
declare -a array=(
    # shellcheck disable=SC2206
    ${FOO}
)
echo "${array[@]}"
```

*Works*
```sh
#!/usr/bin/env bash
true
# shellcheck disable=SC2206
declare -a array=(
    ${FOO}
)
echo "${array[@]}"
```

#### Here's what shellcheck currently says:

```
$ shellcheck myscript
 
Line 5:
    ${FOO}
    ^-- SC2206: Quote to prevent word splitting/globbing, or split robustly with mapfile or read -a.
```

#### Here's what I wanted or expected to see:

Nothing.

**Comment by koalaman** [comment]
ShellCheck actually doesn't have any per-line disabling. All directives are per statement, and not per line. 

This has turned out to be a design mistake, so sorry about that, but as of right now this is intended behavior.
